src/test/test-fs-util.c

```diff
@@ -614,6 +614,77 @@ static void test_fsync_directory_of_file(void) {
         assert_se(fsync_directory_of_file(fd) >= 0);
 }
 
+static void test_rename_noreplace(void) {
+
+        const char* const table[] = {
+                "/reg",
+                "/dir",
+                "/fifo",
+                "/socket",
+                "/symlink",
+                NULL
+        };
+
+        _cleanup_(rm_rf_physical_and_freep) char *z = NULL;
+        const char *e, *j;
+        char **a, **b;
+
+        e = getenv("RENAME_NOREPLACE_DIR");
```

Comment:
I think it'd be nicer to make this an optional positional argument on the command line, but this works too.